src/CMakeLists.txt

```diff
@@ -1,2 +1,4 @@
+add_compile_options(-std=c++17)
```

Comment:
Here a thought. I do not think we have a standard of cpp version right? We usually work with modern c++14/ c++17. I think we should work a bit on this to put some clarity. Even I saw some native C99. C++20 is done and C++23 on its way. Just by curiosity, did you try to build the OSDK with c++20?

Reply:
I did not try it but I think we have to standardize the code as much as we can. Happy to explore C++20/23.